### PR TITLE
feat: support matching files by find_options for http resource

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @hyy0322 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/resolver/http/entrypoint.sh [hunk 1]
```diff
@@ -65,10 +71,17 @@ URL=${URL//'${WORKFLOWRUN_NAME}'/${WORKFLOWRUN_NAME}}
 wrapPush() {
     if [ -z ${COMPRESS_FILE_NAME} ]; then COMPRESS_FILE_NAME="artifact.tar"; fi
     if [ -z ${FORM_FILE_KEY} ]; then FORM_FILE_KEY="file"; fi
+    if [ -z ${FIND_OPTIONS} ]; then FIND_OPTIONS=". -name '*'"; fi
 
-    echo "Start to compress files under ${WORKDIR}/data into file ${COMPRESS_FILE_NAME}"
-    cd ${WORKDIR}/data
-    tar -cvf ${WORKDIR}/${COMPRESS_FILE_NAME} *
+    cd ${WORKDIR}/data/${DATA_SUBDIR}
```
Comment (supereagle) [inline build/resolver/http/entrypoint.sh:76]: This `cd` command is not necessary?

File: build/resolver/http/entrypoint.sh [hunk 2]
```diff
@@ -65,10 +71,17 @@ URL=${URL//'${WORKFLOWRUN_NAME}'/${WORKFLOWRUN_NAME}}
...
+    mkdir -p ${WORKDIR}/__output_resources;
+
+    echo "Start to find and copy files: find ${FIND_OPTIONS} -exec cp --parents {} ${WORKDIR}/__output_resources \;"
+    eval "find ${FIND_OPTIONS} -exec cp --parents {} ${WORKDIR}/__output_resources \;"
+
+    echo "Start to compress files under ${WORKDIR}/data/${DATA_SUBDIR} into file ${COMPRESS_FILE_NAME}"
+    cd ${WORKDIR}/__output_resources
+    tar -cvf ${WORKDIR}/${COMPRESS_FILE_NAME} ./*
```
Comment (supereagle) [inline build/resolver/http/entrypoint.sh:84]:
```suggestion
    tar -cvf ${WORKDIR}/${COMPRESS_FILE_NAME} ${WORKDIR}/__output_resources/*
```

File: build/resolver/http/entrypoint.sh (hunk 1 above)
Comment (zhujian7) [reply build/resolver/http/entrypoint.sh:76]: It is necessary that we need to find and cp files in the ${DATA_SUBDIR}

**Comment by zhujian7** [comment]
@supereagle PTAL

**Pull request merged**